### PR TITLE
Cypress v4

### PR DESCRIPTION
**Pull request by rbclark**
Merging the Cypress v4 branch back into master.

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] Internal ticket for this PR: (N/A)
- [x] Internal ticket links to this PR (N/A)
- [x] Code diff has been done and been reviewed (N/A)
- [x] Tests are included and test edge cases (N/A)
- [x] Tests have been run locally and pass (N/A)

**Reviewer 1:**

Name: @dczulada 
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code 

**Reviewer 2:**

Name: @okeefm
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,5 +1,7 @@
 source 'https://rubygems.org'
 
+ruby '2.3.6'
```
Comment (okeefm) [inline Gemfile]: We should `bundle update ruby` since we bumped the ruby version above.

**Comment by okeefm** [comment]
Do we care about `rubocop` errors? Because there are a lot of them.

**Comment by rbclark** [comment]
@okeefm I was planning to fix both of those errors in the fixingTravis branch.

**Comment by okeefm** [comment]
@rbclark ok, that works

**Comment by rbclark** [comment]
@okeefm @dczulada it looks like someone with admin access will have to force merge this.

**Comment by okeefm** [comment]
@dczulada one of us will have to go in and turn off the status check requirement for this PR. Do you want to do that, or should I?

**Comment by rbclark** [comment]
If either of you are an admin you should be able to force merge and override that requirement.

**Pull request merged**